nucliadb/nucliadb/search/search/merge.py

```diff
@@ -347,6 +353,42 @@ async def merge_paragraph_results(
     )
 
 
+async def merge_relations_results(
+    relations_responses: List[RelationSearchResponse], query: SearchRequest
+) -> Relations:
+    relations = Relations(entities={}, graph=[])
+
+    for entry_point in query.relations.subgraph.entry_points:
+        relations.entities[entry_point.value] = EntitySubgraph(related_to=[])
+
+    for relation_response in relations_responses:
+        for relation in relation_response.subgraph.relations:
+            origin = relation.source.value
+            destination = relation.to.value
+            relation_label = relation.relation_label
+
+            if origin in relations.entities:
+                relations.entities[origin].related_to.append(
+                    DirectionalRelation(
+                        entity=destination,
+                        relation=relation_label,
+                        direction=RelationDirection.OUT,
+                    )
+                )
+            elif destination in relations.entities:
+                relations.entities[destination].related_to.append(
+                    DirectionalRelation(
+                        entity=origin,
+                        relation=relation_label,
+                        direction=RelationDirection.IN,
+                    )
+                )
+            else:
+                assert False, "origin or destination must be in the response"
```

Comment:
This will make the search fail with a 500. Is this intented? Have you considered doing a logger.warning and moving on?

Reply:
It should be improved, but the reason is that if it fails, the node is returning bad results (and should never happen). We can log or send an issue to sentry and continue with the request (makes more sense)

Reply:
I'll change it for a `logger.error` and send an error to Sentry